v0/destinations/iterable/transform.js

```diff
@@ -325,43 +333,202 @@ function process(event) {
+
+    let batchEventResponse = defaultBatchRequestConfig();
+
+    // Batch event into dest batch structure
+    chunk.forEach(ev => {
+      batchResponseList.push(ev.message.body.JSON);
+      metadata.push(ev.metadata);
+    });
+    // batching into bulkUpdate batch structure
+    if (chunk[0].message.endpoint.includes("/api/users")) {
+      batchEventResponse.batchedRequest.body.JSON = {
+        users: batchResponseList
+      };
+      batchEventResponse.batchedRequest.endpoint = IDENTIFY_BATCH_ENDPOINT;
+    } else {
+      // batching into bulkUpdate batch structure
+      batchEventResponse.batchedRequest.body.JSON = {
+        events: batchResponseList
+      };
+      batchEventResponse.batchedRequest.endpoint = TRACK_BATCH_ENDPOINT;
+    }
+
+    batchEventResponse.batchedRequest.headers = {
+      "Content-Type": "application/json",
+      api_key: apiKey
+    };
+
+    batchEventResponse = {
+      ...batchEventResponse,
+      metadata,
+      destination
+    };
+    batchedResponseList.push(
+      getSuccessRespEvents(
+        batchEventResponse.batchedRequest,
+        batchEventResponse.metadata,
+        batchEventResponse.destination,
+        true
+      )
+    );
+  });
+
+  return batchedResponseList;
+}
+
+function getEventChunks(
+  event,
+  identifyEventChunks,
+  trackEventChunks,
+  eventResponseList
+) {
+  // Categorizing identify and track type of events
+  // Checking if it is identify type event
+  if (event.message.endpoint.includes("api/users/update")) {
+    identifyEventChunks.push(event);
+  } else if (event.message.endpoint.includes("api/events/track")) {
+    // Checking if it is track type of event
+    trackEventChunks.push(event);
+  } else {
+    // any other type of event
+    const { message, metadata, destination } = event;
+    const endpoint = get(message, "endpoint");
+
+    const batchedResponse = defaultBatchRequestConfig();
+    batchedResponse.batchedRequest.headers = message.headers;
+    batchedResponse.batchedRequest.endpoint = endpoint;
+    batchedResponse.batchedRequest.body = message.body;
+    batchedResponse.batchedRequest.params = message.params;
+    batchedResponse.batchedRequest.method =
+      defaultPostRequestConfig.requestMethod;
+    batchedResponse.metadata = [metadata];
+    batchedResponse.destination = destination;
+
+    eventResponseList.push(
+      getSuccessRespEvents(
+        batchedResponse.batchedRequest,
+        batchedResponse.metadata,
+        batchedResponse.destination
+      )
+    );
+  }
+}
+
 const processRouterDest = async inputs => {
   if (!Array.isArray(inputs) || inputs.length <= 0) {
     const respEvents = getErrorRespEvents(null, 400, "Invalid event array");
     return [respEvents];
   }
 
-  const respList = await Promise.all(
-    inputs.map(async input => {
+  let identifyEventChunks = []; // list containing identify events in batched format
+  let trackEventChunks = []; // list containing track events in batched format
```

Comment:
As we are mutating these `in place` we should initialise this inside the map function

Reply:
In the map function, we are re-initialising "identifyEventChunks" and "trackEventChunks" only when they reach to respective BATCH_SIZE limit.
If we declare and initialise them within the map function itself, then we might lose existing data in every iteration.

Reply:
Got it we are iterating for every event didn't consider that. Cool got your point